dotenv-linter/src/cli.rs

```diff
@@ -139,6 +139,15 @@ fn common_args() -> Vec<Arg> {
             .long("schema")
             .help("Use schema file to check .env files")
             .value_parser(value_parser!(PathBuf)),
+        Arg::new("stdin")
+            .long("stdin")
+            .help("Read and lint from stdin instead of files")
+            .action(ArgAction::SetTrue),
```

Comment:
Just a note: it still makes sense to support special filename (`-`) passed as it was discussed here: https://github.com/dotenv-linter/dotenv-linter/issues/803#issuecomment-2294979944 (and see the next comment)

But not necessary to mix this in single PR for sure. Just something nice to keep in mind

This may help to build something more aligned, e.g unified handling for `-` and `--stdin`